### ISSUE TITLE
Aw, snap!

### ISSUE THREAD

**Issue by Moriarx** [Codewars/codewars.com#1157]
Hello, 

I cannot gain access to codewars.com pages. I can only get to https://www.codewars.com/users/sign_in
others, to sign up or even I did it on my smartphone are working just fine. I tried everything, IE, Chrome, Firefox, it only works on Safari(on windows, dont have mac) but it's damn slow. I tried to install older versions of Chrome, cleaned cookies, run antyspyware, changed Internet connection but nothing works.

When I press F12 I get this while loading the page:  "" ably.min.js:26 The Ably CDN URL you are using is deprecated because it does not contain a version number. Please see https://support.ably.io/solution/articles/3000063698 ""

However I dont get how to solve this. Anyone please? I'm willing to start coding as soon as possible and this site could help a little bit ;) 
P.S. I'm using Windows 7 64bit

**Comment by kazk** [comment]
On Chrome, what happens when you use Incognito window?
Do you have many tabs open?
Try closing any other applications you have running.
Take a look at <https://support.google.com/chrome/answer/95669> and try things you haven't done.

**Comment by Moriarx** [comment]
I tried that as Tell. Tried closing all tabs, turning off extensions, incognito mode. Only adding linę to chrome -no-sandbox helped but site loaded once, and then it was stuck on like 30% of Full load.

**Comment by kazk** [comment]
Are you using newest Chrome?
What happens if you reload after "Aw, snap!"? Force reload?

I don't know Windows, but using `--no-sandbox` shouldn't be necessary and wouldn't recommend in general.

**Comment by Moriarx** [comment]
I tried with newest chrome and also build from June this year. Force reload makes nothing, again "Aw, Snap" Appears. I cannot find any reasonable solution for this...:(

**Comment by nomennescio** [comment]
I can confirm I had the same issue with a freshly installed Chrome, but are not seeing it on 59.0.3071.86

**Comment by kazk** [comment]
@Moriarx are you still having this issue? I tried Google Chrome  62.0.3202.75 on Windows 10 and I'm unable to reproduce.

![image](https://user-images.githubusercontent.com/639336/32078713-cfa14058-ba5c-11e7-8894-17938eca4d7d.png)
![image](https://user-images.githubusercontent.com/639336/32078782-1263ffac-ba5d-11e7-9d0a-6644ec5b7186.png)

@nomennescio are you also using Windows 7? I'm unable to reproduce this on Windows 10.

**Comment by TheAnton205** [comment]
I have the same problem, I wanted to start Code wars, but it seems to say 'Aw Snap!' every time. I see the page for a second but then i just goes away. I tried everything you guys said, I have the latest chrome, windows 7 running on a 64 bit.

**Comment by yuriko192** [comment]
it's a problem because you're using a windows 7. when i upgraded the system, i could open the webpage. seems like the web isn't optimized for all OS

**Comment by khovansky-al** [comment]
Chrome 61.0.3163.100 Windows 8.1 has the same problem

**Comment by khovansky-al** [comment]
Successfully logged in by clearing cookies, going to login page directly and re logging-in there. Maybe this was not necessary and it's only the main page that is not working. Dashboard and others seem to be fine.

**Comment by khovansky-al** [comment]
Nah, disregard previous comment. It crashes in the new Firefox as well, and I can't figure a pattern. It just does.

**Comment by dmkach** [comment]
Have the same issue on windows 8.1 using Chrome Version 62.0.3202.94 (Official Build) (64-bit)
In the Firefox it's working fine.

**Comment by crosby-lv** [comment]
After I installed KB3125574,the issue is resolved,it seem the operator system's problem

**Comment by mhmd9y** [comment]
as @crosby-lv  said, I think it's something related to the OS not codewars at all.  
I had the same issue with Windows 8.1, so I went to control panel checked the available updates and installed `KB4054519`.  
It all works fine now.

**Comment by avernikoz** [comment]
I have the same issue on my win 7 sp1 laptop, so, if I use old version of firefox (39.0) everything works fine.

**Comment by ProductiveCode** [comment]
I also have the same issue with firefox, Opera, SRWare Iron and chrome on windows 7.  Incognito or not.

**Comment by ak1nak** [comment]
Same here.
Can not open in any browser
Only page i can get is https://www.codewars.com/users/sign_in
In chrome  65.0.3325.146  im gettin https://pbs.twimg.com/profile_images/378800000406239890/084577ff6703d0fae39cd9d9c869e607_bigger.jpeg 404 ()  (index):27

same in incognito tab but w/o any errors

cant open https://www.codewars.com/ in firefox latest version either

**Comment by sunlazor** [comment]
Same issue
Firefox 59.0 (64-bit) in safe mode
Chromium  55.0.2883.87 (64-bit)
https://www.codewars.com/users/sign_in - ok
https://www.codewars.com/dashboard - ok
https://www.codewars.com/users/{user} - ok
https://www.codewars.com/trainer/setup -  crash
https://www.codewars.com/users/{user}/completed - crash

**Comment by tair-alisher** [comment]
chromium - crash
firefox - crash
vivaldi - ok

**Comment by Leeoku** [comment]
had w7 ultimate sp1. Couldn't get any of the browsers to work, incognito, and clearing cookies. Updating windows made it work

**Comment by amllr** [comment]
Got same problem win 7 x64 sp1 Chrome 65.0.3325.181 and  Firefox 59.0.2 both
Problem perfectly solved using SRWare Iron 65.0.3400.0 32-bit  and also of course Tor Browser 7.5.3 worked.
--

**Comment by aryan-firouzian** [comment]
I am also facing the same issue on Chrome and Firefox. Dashboard page opens without any problem, but "http://www.codewars.com/kata/my-languages" page result in "Aw, snap!", I only checked it on my personal laptop, and not any other device.
Chrome Version 66.0.3359.139 (Official Build) (64-bit)
Firefox Version 59.0.2 (64-bit)

**Comment by aryan-firouzian** [comment]
I also tried on Opera Version 53.0.2907.37, and same problem happened. It seems that it is related to the OS (mine is also windows 7 64bit), but it is interesting that only happens to codewars website.

**Comment by neltharionorg** [comment]
![43](https://user-images.githubusercontent.com/28743090/39968927-e3e2f004-56e5-11e8-8288-a58824bb32df.PNG)
![capture](https://user-images.githubusercontent.com/28743090/39968928-e4058178-56e5-11e8-8301-34d82caf7019.PNG)

**Comment by teekcode** [comment]
visit this https://www.codewars.com/users/sign_in if you want to login in and coding. just for a temporary solution.

**Comment by ak1nak** [comment]
> visit this https://www.codewars.com/users/sign_in if you want to login in and coding. just for a temporary solution.
Yea, but if I'm a new user? https://www.codewars.com/join not working as well.

**Comment by darsh97** [comment]
I am facing this issue now. The Dashboard runs fine on Chrome, the rest crashes.

**Comment by ngomile** [comment]
Just adding on to this but I'm experiencing the same thing on Firefox 60. Landing on the homepage leads to an almost immediate crash. But here's a weird thing that's happening, when I go to https://www.codewars.com/kata/latest it still crashes. But if I open my toolbox and reload the page it doesn't crash. But once I close the toolbox on that page it immediately crashes.

**Comment by kikit** [comment]
I'm using Chrome Version 69.0.3497.92 (Official Build) (64-bit) and latest version of Firfox as well. Site keep crashing.

![image](https://user-images.githubusercontent.com/6348575/45564714-f324f900-b86e-11e8-9489-0a3c55343295.png)

![image](https://user-images.githubusercontent.com/6348575/45564746-0f289a80-b86f-11e8-84e6-fbdbcbc23b75.png)

**Comment by thaumocracy** [comment]
Still crashing for about half a year. Win7,only working browser for me is Yandex browser.

**Comment by alexamy** [comment]
ugh, same problem, win7 and chrome 69.0.3497.100 64bit. also in firefox

**Comment by pbaveja** [comment]
**I think I figured out a temporary fix.** 
There seems to be an issue with the **icon asset named icomoon.** I don't know if its codewars asset that's causing the problem, or if Chrome/other browsers cannot render it.

I installed a request blocker extension(screenshot below) and put in *://www.codewars.com/assets/icomoon* to not load this asset, and all pages work fine. Although, of course, some icons don't load.
![clipboard01](https://user-images.githubusercontent.com/22319179/46794021-941aad00-cd64-11e8-85fe-9a042edcc1eb.jpg)

**Comment by alexamy** [comment]
@pbaveja's solution is working for me too

**Comment by alexamy** [comment]
iconmoon issue also referenced here: https://github.com/Codewars/codewars.com/issues/1003#issuecomment-361018536

**Comment by mzmb** [comment]
It seems the problem appears while rendering _Kotlin_ language icon. You can prevent Chrome from doing this, for example, by adding a custom filter for _uBlock Origin_ extension: 
```
!Prevents CodeWars from crashing
www.codewars.com##i.icon-moon-kotlin
```
Now there is a blank space instead of ![30523722-0101ba56-9bef-11e7-929c-8843369c5d44](https://user-images.githubusercontent.com/17899306/48370358-28609280-e6ca-11e8-941d-db567f0a289f.png), but everything else works fine.

**Comment by Brollof** [comment]
@mzmb 
Main page https://www.codewars.com/dashboard works fine with this workaround but this one still doesn't: https://www.codewars.com/kata/latest/my-languages (Aw snap! error).

**Comment by alexamy** [comment]
@mzmb workaround is working perfectly for me
@Brollof https://www.codewars.com/kata/latest/my-languages works fine

**Comment by coolpage5** [comment]
I have the same problem, I wanted to start Code wars, but it seems to say 'Aw Snap!' every time. Win 8.1, browsers Chrome 71, Firefox 64.0.2
But in YandexBrowser v 18.11.1.805 it's work! ))

**Comment by Voileexperiments** [comment]
@kazk I investigated this a bit further and found that the `.icon-moon-kotlin` (codepoint e903) character indeed has problems, but only on specific size range. Looking at [the svg file](view-source:https://www.codewars.com/assets/icomoon-c39dfe742aa7f19d9a7ab18f28b60841.svg#ico-moon) I suspect that it's due to the (idiotic) way the svg is drawn: to emulate diagonal lines there were *lots* of `h0.954v-0.954`s. In fact when displaying the glyph from the original svg or from that converted from tff (using one of the online conversion services), the glyph is flipped upside down for some reasons. :man_facepalming: 

I then tried to convert *this* svg into ttf and injected the converted font into the css, which finally stops the crash, but the kotlin font is still not rendered at the zoom range that caushed crashes before.

I then looked at the icomoon icon list and located the Kotlin icon (which is a *sane* version of it): https://simpleicons.org/?q=kotlin

Replacing the svg path to this icon's svg path indeed makes the conversion work, and then I used [this](https://stackoverflow.com/a/35354087) to transform the svg to roughly match what I've got from the (broken) original CW svg file:

```M 116.23 892.4 l 419.23 -426.65 L 958.4 892.4 z M 68 2 h 445.2 L 68 465.75 z M 565.14 2 L 68 521.4 v 371 l 445.2 -445.2 L 958.4 2 z```

And then plugged in this, generated the tff and injected this fixed tff into the css, and the problem is fixed.

In short: I think the icomoon generated font files needs to be updated ;-)

---

I used this setup to debug the issue which you might find useful:

1. Chrome with Stylus extension to override user agent stylesheet with the exact content of CW's CSS
2. Use request blocking to block CW's own CSS
3. I can inject my own font files by changing the `@font-face` block in the CSS and test them

**Comment by kazk** [comment]
@Voileexperiments thank you so much for doing this! I'll try updating it soon.

**Comment by kazk** [comment]
I've updated the font files. Thanks again @Voileexperiments.

**Comment by Voileexperiments** [comment]
@kazk I can see the changes in preview site but not on production site. I don't think the fonts have been deployed to it, or do we have to wait for a while before the CDN catches the new files?

**Comment by kazk** [comment]
@Voileexperiments I forgot to purge the cache for production. I just did it. Can you see it now?

**Comment by kazk** [comment]
I was using the wrong url when purging. I just cleared everything so it should start working.

**Comment by Voileexperiments** [comment]
Can confirm it's applied to production now.
